### PR TITLE
ENH glibc 2.17

### PR DESCRIPTION
**Pull request by beckermr**
Checklist
* [ ] Used a [fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by beckermr** [comment]
SHAs are wrong. Will fix.

**Comment by isuruf** [comment]
Can you add support for aarch64 and ppc64le too?

**Comment by beckermr** [comment]
Yup happy to add.

**Comment by beckermr** [comment]
@conda-forge-admin rerender

**Comment by beckermr** [comment]
@conda-forge-admin rerender

**Comment by beckermr** [comment]
@isuruf Any idea why am I not getting builds on aarch64 here?

**Comment by isuruf** [comment]
Because this is a `noarch: generic` package and CDTs for aarch64 are built on `linux-64`.

**Comment by beckermr** [comment]
Ahhh. I need to template with jinja

**Comment by isuruf** [comment]
@conda-forge-admin, rerender

**Comment by beckermr** [comment]
@conda-forge-admin rerender

**Comment by beckermr** [comment]
@conda-forge-admin rerender

**Comment by beckermr** [comment]
OK @isuruf This one is green. Note there is a slight difference in how `sysroot/urs/include/gnu/stubs.h` is done on aarch64 versus the other platforms, so I had to adjust the test. I think this is ok, but it would be good if you can double-check it.

**Comment by beckermr** [comment]
Ack. Before we merge, do we need to put in a virtual package constraint?

**Comment by beckermr** [comment]
We should also mark the 2.12 version as broken since nothing is using it and it doesn't have that constraint.

**Comment by beckermr** [comment]
@conda-forge-admin rerender

File: recipe/meta.yaml [hunk 1]
```diff
@@ -30,24 +52,29 @@ build:
...
 
 outputs:
-  - name: sysroot_{{ target_platform }}
+  - name: sysroot_{{ cross_target_platform }}
     script: build-sysroot.sh
     build:
       noarch: generic
+    requirements:
+      run_constrained:
+        - __glibc >={{ version }},<3.0.a0
```
Comment (isuruf) [inline recipe/meta.yaml:61]: Why do we need this constraint?

Comment (beckermr) [reply recipe/meta.yaml:61]: If we build packages using 2.17, will they work in machines that locally have an earlier version?

Comment (isuruf) [reply recipe/meta.yaml:61]: No, but then we need a `run_exports` not a `run_constrained`.

Comment (beckermr) [reply recipe/meta.yaml:61]: But this is a CDT. It doesn’t ever appear in an env at runtime. It is just used for building.

Comment (beckermr) [reply recipe/meta.yaml:61]: We link at runtime to the local copy of glibc, right?

Comment (beckermr) [reply recipe/meta.yaml:61]: The virtual package is a constraint on this local version.

Comment (isuruf) [reply recipe/meta.yaml:61]: > It doesn’t ever appear in an env at runtime.

Then what is the use of this `run_constrained`? This only prevents users from cross compiling.

File: recipe/meta.yaml [hunk 2]
```diff
@@ -30,24 +52,29 @@ build:
     - '*'
 
 outputs:
-  - name: sysroot_{{ target_platform }}
+  - name: sysroot_{{ cross_target_platform }}
     script: build-sysroot.sh
     build:
       noarch: generic
```
Comment (isuruf) [inline recipe/meta.yaml:58]: Can you add a `track_features` entry here?

File: recipe/meta.yaml (hunk 1 above)
Comment (beckermr) [reply recipe/meta.yaml:61]: Ohhhhhhh. You mean a run export of the virtual package constraint, not of the cdt. Facepalm. Yes that will work but ppl with conda < 4.8 will have trouble.

File: recipe/meta.yaml (hunk 2 above)
Comment (beckermr) [reply recipe/meta.yaml:58]: Should the cdt version be in it?

File: recipe/meta.yaml (hunk 1 above)
Comment (isuruf) [reply recipe/meta.yaml:61]: Sure. That's okay in my opinion. If they want new things, use newer conda.

File: recipe/meta.yaml (hunk 2 above)
Comment (isuruf) [reply recipe/meta.yaml:58]: Yes

**Comment by beckermr** [comment]
@conda-forge-admin rerender

**Comment by beckermr** [comment]
@conda-forge-admin rerender

**Comment by beckermr** [comment]
ok @isuruf - this is ready for another look.

I went ahead and marked the 2.12 version of this broken since we will need to extract the 2.12.2 version from the ctng compilers anyway and it is missing the exports. We could patch the repo data, but this seems like a pain and not worth it since we are not using it anyways.

**Comment by isuruf** [comment]
Please don't. We don't need a `run_exports`. We can even patch the stdlib.h for that `aligned_alloc` thing.

**Comment by beckermr** [comment]
OK. back on main. You ready to merge this?

**Comment by isuruf** [comment]
Thanks.

**Pull request merged**